proof-systems/src/darlin/pcd/simple_marlin.rs

```diff
@@ -23,6 +23,143 @@ use std::marker::PhantomData;
+    max_domain_h_size: usize,
+    max_domain_k_size: usize,
+) -> bool
+{
+    let segment_size: usize = 1 << proof.pc_proof.proof.l_vec.len();
+    let zk_bound = if proof.pc_proof.proof.hiding_comm.is_some() { 1 } else { 0 };
+    let w_z_a_b_segs = ((max_domain_h_size + 2 * zk_bound) as f64/segment_size as f64).ceil() as usize;
+    let t_segs = ((max_domain_h_size as f64/segment_size as f64)).ceil() as usize;
+    let z_1_segs = ((max_domain_h_size + 3 * zk_bound) as f64/segment_size as f64).ceil() as usize;
+    let h_1_segs = ((2 * max_domain_h_size + 4 * zk_bound - 2) as f64/segment_size as f64).ceil() as usize;
+    let z_2_segs = (max_domain_k_size as f64/segment_size as f64).ceil() as usize;
+    let h_2_segs =  ((3 * max_domain_k_size - 3) as f64/segment_size as f64).ceil() as usize;
+
+    is_vk_consistent_with(vk, segment_size, max_domain_h_size, max_domain_k_size) &&
+        segment_size >= min_segment_size &&
+        segment_size <= max_segment_size &&
+        proof.commitments[0][0].comm.len() <= w_z_a_b_segs && // w poly
+        proof.commitments[0][1].comm.len() <= w_z_a_b_segs && // z_a poly
+        proof.commitments[0][2].comm.len() <= w_z_a_b_segs && // z_b poly
+        proof.commitments[1][0].comm.len() <= t_segs && // t poly
+        proof.commitments[1][1].comm.len() <= z_1_segs && // z_1 poly
+        proof.commitments[1][2].comm.len() <= h_1_segs && // h_1 poly
+        proof.commitments[2][0].comm.len() <= z_2_segs && // z_2 poly
+        proof.commitments[2][1].comm.len() <= h_2_segs // h_2 poly
+}
+
+/// Given segment_size, density, zk, proof_type, return the maximum value of |H| s.t. proof size will be <= max_proof_size
+/// and the corresponding proof size and vk size
+pub fn compute_max_domain_h_size(
+    segment_size: usize,
+    density: usize,
+    zk: bool,
+    max_proof_size: usize,
+    proof_type: &str
+) -> (usize, usize, usize)
+{
+    let zk_bound: usize = if zk { 1 } else { 0 };
+    let segment_size = segment_size.next_power_of_two();
+    let mut max_supported_h_size = 0;
+    let mut max_supported_proof_size = 0;
+    let mut max_supported_vk_size = 0;
+
+    loop {
+        let h = 1 << max_supported_h_size;
+        let k = (h * density).next_power_of_two(); // |K|/|H| = density
+        let w_z_a_b_segs = ((h + 2 * zk_bound) as f64/segment_size as f64).ceil() as usize;
+        let t_segs = ((h as f64/segment_size as f64)).ceil() as usize;
+        let z_1_segs = ((h + 3 * zk_bound) as f64/segment_size as f64).ceil() as usize;
+        let h_1_segs = ((2 * h + 4 * zk_bound - 2) as f64/segment_size as f64).ceil() as usize;
+        let z_2_segs = (k as f64/segment_size as f64).ceil() as usize;
+        let h_2_segs =  ((3 * k - 3) as f64/segment_size as f64).ceil() as usize;
+
+        if w_z_a_b_segs > 255 || t_segs > 255 || z_1_segs > 255 || h_1_segs > 255 || z_2_segs > 255 || h_2_segs > 255 {
+            max_supported_h_size += 1;
+            continue;
+        }
```

Comment:
Given a fixed segment size, why do we INCREASE ``max_supported_h_size`` if one of the polynomials already exceeds the sanity bound of 255 segments? For higher domain sizes this clause will be triggered, too.